pandas/core/groupby/generic.py

```diff
@@ -1664,3 +1655,36 @@ def _normalize_keyword_aggregation(kwargs):
         order.append((column,
                       com.get_callable_name(aggfunc) or aggfunc))
     return aggspec, columns, order
+
+
+def _recast_datetimelike_result(result):
+    """
+    If we have date/time like in the original, then coerce dates
+    as we are stacking can easily have object dtypes here.
+
+    Parameters
+    ----------
+    result : DataFrame
+
+    Notes
+    -----
+    - Assumes Groupby._selected_obj has ndim==2 and at least one
```

Comment:
this note doesn't seem relevant as you are passing in frame right?

Reply:
That wasn't obvious to me bc were talking about the dimensions of two separate objects.  Are they necessarily the same?